sample-compose/app/src/main/java/co/nimblehq/sample/compose/ui/screens/home/HomeComposeViewModel.kt

```diff
@@ -33,10 +41,31 @@ class HomeComposeViewModel @Inject constructor(
                     _uiModels.emit(uiModels)
                 }
             hideLoading()
+
+            requestPermissions()
         }
     }
 
     fun navigateToSecond(uiModel: UiModel) {
         execute { _navigator.emit(NavigationEvent.Second(uiModel)) }
     }
+
+    private suspend fun requestPermissions() {
+        _requestPermissions.emit(
+            arrayOf(
+                Manifest.permission.ACCESS_FINE_LOCATION,
+                Manifest.permission.CAMERA
+            )
+        )
+    }
+
+    fun onPermissionResult(permissionResult: PermissionResult) {
+        when (permissionResult) {
+            is Granted -> Timber.d("${permissionResult.grantedPermissions} granted")
+            is Denied.JustDenied -> Timber.d("${permissionResult.deniedPermissions} denied")
+            is Denied.NeedsRationale -> Timber.d("${permissionResult.deniedPermissions} needs rationale")
+            is Denied.DeniedPermanently -> Timber.d("${permissionResult.deniedPermissions} denied for good")
+            is Cancelled -> Timber.d("request cancelled")
```

Comment:
@ryan-conway Any idea how to trigger `Cancelled` & `JustDenied`? I can trigger the rest correctly though

Reply:
According the the [readme for v2.1.2](https://github.com/deva666/Peko/tree/release/2.1.2)
>PermissionResult.Denied.JustDenied -> subclass of PermissionResult.Denied, returned when previous two cases are not the cause, for example if you forget to register the Permission in AndroidManifest
PermissionResult.Cancelled -> returned when Android System cancels the request, ie returned

Reply:
Nice, thank you 🙏 

Also, when I tap outside the permission dialog, it results in `DeniedPermanently`.

Not what I was expecting... What do you think? 💭

Reply:
Looks like this is an [existing issue](https://github.com/deva666/Peko/issues/21) with the library 👀 
Not sure if this has been fixed in the latest version, but even if it was we can't use it anyway until we update our dependencies 🥲 
Given that, shall we still continue to use the library? Seems like a pretty big issue to have considering this is meant to be a good template to use 😅

Reply:
Ughhhh, that sucks 😭 

I'm afraid we'll have to put this on hold for now and wait until we can upgrade to `SDK 31` (Probably in December)

Hopefully, the issue has been fixed by then.

If you agree, you can put this pull request as a draft 🪦

Reply:
Agreed, will draft it for now 🥲